src/sqlfluff/core/dialects/base.py

```diff
@@ -196,6 +154,19 @@ def replace(self, **kwargs: DialectElementType):
                 raise ValueError(f"{n!r} is not already registered in {self!r}")
             self._library[n] = kwargs[n]
 
+    def add_update_segments(self, module_dct):
+        """Scans module dictionary, adding or replacing segment definitions.
+
+        Each dialect module (.py file) should call this function at the END OF
+        THE FILE, after all the segment classes have been defined.
```

Comment:
Anyway we could automate this so we don't have to add? e.g. by adding to `load_raw_dialect`?

Reply:
Good call -- calling it from `load_raw_dialect()` seems to be working fine. I'll update.